### PR TITLE
feat(start_planner): apply offset colission check section

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

this [PR](https://github.com/autowarefoundation/autoware.universe/pull/5952) should be merged first

offset the collision check section based on shift end pose.
collision check against static objects are executed in the collision check section.

image

ignore section's lenght is controlled by parameter of 
`collision_check_distance_from_end `

## Tests performed

- [x] [PASS TIERIV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/fdfaf13b-b665-501c-9e17-baa48738e21f?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 1]
```diff
@@ -936,18 +948,27 @@ std::vector<Pose> StartPlannerModule::searchPullOutStartPoseCandidates(
   return pull_out_start_pose_candidates;
 }
 
-PredictedObjects StartPlannerModule::filterStopObjectsInPullOutLanes(
-  const lanelet::ConstLanelets & pull_out_lanes, const double velocity_threshold) const
+PredictedObjects StartPlannerModule::filterStopObjectsInShoulderLanes(
+  const lanelet::ConstLanelets & shoulder_lanes, const geometry_msgs::msg::Point & current_point,
+  const double velocity_threshold, const double object_check_forward_distance,
+  const double object_check_backward_distance) const
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:954]: pull out lanes will be road lanes not just shoulder lanes.

Comment (kosuke55) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp:954]: so, better to change function and args name

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 2]
```diff
@@ -676,6 +660,31 @@ bool StartPlannerModule::findPullOutPath(
...
+  // calculate collision check end idx
+  const size_t collision_check_end_idx = std::invoke([&]() {
+    const auto collision_check_end_pose = motion_utils::calcLongitudinalOffsetPose(
+      combined_path.points, path.end_pose.position, parameters_->collision_check_distance_from_end);
+
+    if (collision_check_end_pose) {
+      return motion_utils::findNearestIndex(
+        combined_path.points, collision_check_end_pose->position);
+    } else {
+      return combined_path.points.size() - 1;
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:679]:
```suggestion
      return 0;
```
is reasonable?

Comment (kosuke55) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp:679]: or nullopt?

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp (hunk 1 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp]: changed in https://github.com/autowarefoundation/autoware.universe/pull/5982/commits/365b4f90f2701f4da883b743acd7913630da5349

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp (hunk 2 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp]: changed in https://github.com/autowarefoundation/autoware.universe/pull/5982/commits/6839191456f0dc84228f824dd4dbc692d48d20b3

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 3]
```diff
@@ -937,16 +951,25 @@ std::vector<Pose> StartPlannerModule::searchPullOutStartPoseCandidates(
...
+  const lanelet::ConstLanelets & pull_out_lanes, const geometry_msgs::msg::Point & current_point,
+  const double velocity_threshold, const double object_check_forward_distance,
+  const double object_check_backward_distance) const
 {
   const auto stop_objects = utils::path_safety_checker::filterObjectsByVelocity(
     *planner_data_->dynamic_object, velocity_threshold);
 
-  // filter for objects located in pull_out_lanes and moving at a speed below the threshold
-  const auto [stop_objects_in_pull_out_lanes, others] =
+  // filter for objects located in shoulder lanes and moving at a speed below the threshold
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:961]:
```suggestion
  // filter for objects located in pull out lanes and moving at a speed below the threshold
```

**Review by kosuke55 — APPROVED** [review]
LGTM!!

**Pull request merged**